runtime/common/src/apis.rs

```diff
@@ -420,10 +420,17 @@ macro_rules! impl_runtime_apis_plus_common {
 							// return false if author mapping not registered like in can_author impl
 							return false
 						};
+						let candidates = pallet_parachain_staking::Pallet::<Self>::compute_top_candidates();
+						if candidates.is_empty() {
+							// If there are zero selected candidates, we use the same eligibility
+							// as the previous round
+							return AuthorInherent::can_author(&author, &slot);
+						}
+
```

Comment:
This code branch devolves into the same one at line 438 (for non-round-change-boundaries).

Could we rewrite the whole thing to be something simpler, like:

```rust
if AuthorInherent::can_author(&author, &slot) {
    true
} else if /* is block round change */ {
    // attempt prediction
}
```

Reply:
Maybe, but not like that because `AuthorInherent::can_author` can return true for a candidate that would not be eligible after the round change.